File: hackerrank/30 days of code/day 25, running time and complexity/day_25_running_time_and_complexity.py
import math

N = int(input())

for _ in range(N):
    number = int(input())

    if number == 1:
        print("Not prime")
        continue

    i, limit = 2, math.sqrt(number)
    while i <= limit:
        if not (number % i):
            print("Not prime")
            break
        i += 1

    if i > limit:
        print("Prime")

File: hackerrank/30 days of code/day 25, running time and complexity/day_25_running_time_and_complexity.cs
using System;

class Solution {
    static void Main(String[] args) {        
        for (var n = Int32.Parse(Console.ReadLine()); n-- > 0; ) {
            var number = Int32.Parse(Console.ReadLine());

            if (number == 1) {
                Console.WriteLine("Not prime");
                continue;
            }

            int i = 2, limit = (int)Math.Sqrt(number);
            for (; i <= limit; i++)
                if (number % i == 0) {
                    Console.WriteLine("Not prime");
                    break;
                }

            if (i > limit)
                Console.WriteLine("Prime");
        }
    }
}
